resolver-dns/src/main/java/io/netty/resolver/dns/BiDnsQueryLifecycleObserver.java

```diff
@@ -0,0 +1,96 @@
+package io.netty.resolver.dns;
+
+import io.netty.channel.ChannelFuture;
+import io.netty.handler.codec.dns.DnsQuestion;
+import io.netty.handler.codec.dns.DnsResponseCode;
+import io.netty.util.internal.UnstableApi;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+/**
+ * Combines two {@link DnsQueryLifecycleObserver} into a single {@link DnsQueryLifecycleObserver}.
+ */
+@UnstableApi
+public final class BiDnsQueryLifecycleObserver implements DnsQueryLifecycleObserver {
```

Comment:
> You only need to create the list one time when create the factory and the pass the same instance to each observer

The me clarify to make sure we are on the same page...

There is `BiDnsQueryLifecycleObserverFactory` which makes it so 2 `DnsQueryLifecycleObserverFactory` can be used at the same time. `BiDnsQueryLifecycleObserverFactory#newDnsQueryLifecycleObserver` will be called on every DNS query, and this must call both factories to generate two new `DnsQueryLifecycleObserver`s and pass these to a new `BiDnsQueryLifecycleObserver`. We would need to create a list for the factory one time, but then on every query we would need to create a list to give to `BiDnsQueryLifecycleObserver` ... creation of a list on every query is what I want to avoid.